internal/checker/checker.go

```diff
@@ -17993,8 +18003,9 @@ func (c *Checker) addInheritedMembers(symbols ast.SymbolTable, baseSymbols []*as
 func (c *Checker) resolveDeclaredMembers(t *Type) *InterfaceType {
 	d := t.AsInterfaceType()
 	if !d.declaredMembersResolved {
+		members := c.getMembersOfSymbol(t.symbol)
 		d.declaredMembersResolved = true
-		d.declaredMembers = c.getMembersOfSymbol(t.symbol)
+		d.declaredMembers = members
```

Comment:
This change enables recursion through `getMembersOfSymbol` which we rely on to first resolve early-bound members so they can be referenced in resolution of late-bound members.